tests/container/build.sh

```diff
@@ -150,7 +203,23 @@ export WRITABLE_PASSWD=yes
 export NSS_WRAPPER_PASSWD=/etc/passwd
 export NSS_WRAPPER_GROUP=/etc/group
 
+# when called with --install as the first argument, go into a special mode
+# typically used to just install the container's dependency packages
+if [[ "$1" = "--install" ]]; then
+    task_sys_deps
+    exit $?
+fi
+
+# if critical packages (currently just git) are missing we assume that
+# we need to automatically enable the task_sys_deps step.
+# this step is not enabled by default due to the overhead that updating
+# the dnf repos creates on the overall build time.
+if ! command -v git &>/dev/null ; then
```

Comment:
Isn't this equivalent to checking `SAMBACC_MINIMAL`?

Reply:
SAMBACC_MINIMAL is there to create a container that is basically just a base layer (os) + a layer containing the build script itself  (no build dependencies). This block is typically triggered *within* a minimal container OR in the case that the build script is copied from one container to another to install all the missing dependencies.  The idea here is that a minimal container can just act as a fancy bundle for including the build.sh in other contexts.  Example:

(untested & simplified but you should get the idea)
```
FROM sambacc:minimal AS sambacc
FROM quay.io/centos/centos:stream9 AS builder
COPY --from=sambacc /usr/local/bin/build.sh /tmp/build.sh
# actually build sambacc on centos
RUN /tmp/build.sh {...other args...}
FROM quay.io/centos/centos:stream9
RUN dnf install deps
COPY --from=builder /var/tmp/build/*.rpm  /tmp/rpms/
RUN dnf install sambacc
```

Reply:
May be I am not grasping it completely but I had a different thought of changing `SAMBACC_MINIMAL` to an environment variable(`ENV`) while building sambacc container and to act based on its value instead of checking for _git_ here. This is under an assumption that _git_ won't be included in any of the base OS images.